### PR TITLE
Add pre-onboarding introduction screen

### LINKED ISSUE #172: feat: add pre-onboarding and start tutorial

### PR DESCRIPTION
**Pull request by pmdlt**
## Description

This PR introduces a new pre-onboarding screen to enhance the user experience for new users of ActualIA. The pre-onboarding screen is designed to provide an introductory walkthrough of the app's main features before users proceed to set their news preferences.

### Changes Made
- Added three new onboarding images (`onboarding1.png`, `onboarding2.png`, `onboarding3.png`) to the assets.
- Created a new `pre-onboarding_view.dart` file to define the pre-onboarding screen layout and logic.
- Updated `main.dart` to display the pre-onboarding screen when the app is launched for the first time.
- Modified `themes.dart` to include new theme colors for the pre-onboarding screen.
- Updated `pubspec.yaml` and `pubspec.lock` to include the `introduction_screen` package, which is used to create the onboarding experience.

### New Features
- **Pre-Onboarding Screen**: Introduces the app to new users with a series of three slides highlighting key features:
  1. A brief overview of ActualIA.
  2. Description of the personalized news feed.
  3. Information on how to stay informed with detailed news sources.
- **Navigation**: Users can skip or complete the introduction to proceed to the interest wizard screen.

> **Important :**
> The 3 screens are PoCs, the point of this PR is to add functionality, not to decide on the content of the texts (which will be done at the meeting/verbally). If you wish to modify the content of one of the texts, you can still make a change proposal (easily acceptable via button) or commit to this branch to add/modify a screen.

## Related Issue
Closes #172

## Checklist
- [x] Added new assets for onboarding images.
- [x] Created and styled the pre-onboarding screen.
- [x] Integrated the pre-onboarding screen with the existing app flow.
- [x] Updated dependencies to include `introduction_screen`.

### CODE REVIEW

File: lib/views/pre-onboarding_view.dart [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+              const InterestWizardView()), // For testing purposes, we can replace this with HomePage()
+    );
+  }
+
+  Widget _buildImage(String assetName, [double width = 250]) {
+    return Image.asset('assets/img/$assetName', width: width);
+  }
+
+  @override
+  Widget build(BuildContext context) {
```
Comment (pmdlt) [inline lib/views/pre-onboarding_view.dart:30]: This data comes from `utils/theme.dart`, I didn't manage to import it properly after 45min fighting with it. I'll probably modify `theme.dart` before the merge to export the values.

File: lib/views/pre-onboarding_view.dart [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        height: 1.4,
+        fontWeight: FontWeight.w300);
+
+    const pageDecoration = PageDecoration(
+      titleTextStyle: titleSmall,
+      bodyTextStyle: bodyTextStyle,
+      bodyPadding: EdgeInsets.fromLTRB(16.0, 0.0, 16.0, 16.0),
+      pageColor: THEME_LIGHTGRAY,
+      imagePadding: EdgeInsets.zero,
+    );
```
Comment (Thechi2000) [inline lib/views/pre-onboarding_view.dart:48]:
```suggestion
var pageDecoration = PageDecoration(
      titleTextStyle: Theme.of(context).textTheme.titleMedium!,
      bodyTextStyle: Theme.of(context).textTheme.bodyMedium!,
      bodyPadding: const EdgeInsets.fromLTRB(16.0, 0.0, 16.0, 16.0),
      pageColor: THEME_LIGHTGRAY,
      imagePadding: EdgeInsets.zero,
    );
```

This seems to do the trick ;)

File: lib/views/pre-onboarding_view.dart [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+    return IntroductionScreen(
+      key: introKey,
+      globalBackgroundColor: THEME_LIGHTGRAY,
+      allowImplicitScrolling: true,
+      autoScrollDuration: 3000,
+      infiniteAutoScroll: true,
```
[the remark this answers is not in the document]
Comment (pmdlt) [reply lib/views/pre-onboarding_view.dart:55]: True, I left it to show the possibility, but I was expecting a comment like that. I'm turning it off!

File: lib/views/pre-onboarding_view.dart (hunk 2 above)
Comment (pmdlt) [reply lib/views/pre-onboarding_view.dart:48]: Wtf it didn't compile when I tried this method. Seems to work now, thanks!

File: lib/views/pre-onboarding_view.dart [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+            ElevatedButton(
+              onPressed: () => _onBackToIntro(context),
+              child: const Text('Back to Introduction'),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
```
[the remark this answers is not in the document]
Comment (pmdlt) [reply lib/views/pre-onboarding_view.dart:135]: When I mentioned testing in the comment, I meant "production testing". You can easily test the return to onboarding by hijacking via this view

[the remark this answers is not in the document]
Comment (Thechi2000) [reply lib/views/pre-onboarding_view.dart]: Oh, right. To get the view, I overrode the `isOnboardingRequired` in the `AuthModel` to always be true.

**Review by Thechi2000 — APPROVED** [review]
LGTM ! Thanks for the work :)

**Review by Pacsort17 — APPROVED** [review]
Tested on Android, works fine.

Tests could be added to detect the small arrow and swipe if you have time to.

**Pull request merged**